code/modules/mechs/mech_life.dm

```diff
@@ -67,13 +67,18 @@
 /mob/living/exosuit/handle_environment(var/datum/gas_mixture/environment)
 	if(!environment) return
 	//Mechs and vehicles in general can be assumed to just tend to whatever ambient temperature
-	if(abs(environment.temperature - bodytemperature) > 10 )
-		bodytemperature += ((environment.temperature - bodytemperature) / 3)
-
-	if(environment.temperature > material.melting_point * 1.25 ) //A bit higher because I like to assume there's a difference between a mech and a wall
```

Comment:
Main issue was that bodytemp was ignored, also damage ticks could be in the many hundreds.